src/bootloader/bootloader.c

```diff
@@ -766,6 +768,16 @@ static size_t _api_screen_rotate(uint8_t* output)
     return _report_status(OP_STATUS_OK, output);
 }
 
+static size_t _api_hardware(uint8_t* output)
+{
+    uint8_t type = 0;
+    if (memory_get_securechip_type() == MEMORY_SECURECHIP_TYPE_OPTIGA) {
+        type = 1;
+    }
+    memcpy(output + BOOT_OP_LEN, &type, 1);
+    return BOOT_OP_LEN + 1;
```

Comment:
The second byte of the output for all api commands is the success/error byte, but it's missing here. The bootloader `._query()` call checks the second byte for error. It seems that you put the hardware type into this byte, so the Python client does not actually work as expected.

You should add it and set it by using `return _report_status(result, output) + ...;`

The memcpy of a single byte is a bit awkward. You can ust do `output[2] = type`

Reply:
_api_versions is also a bit borked: returns the size of three versions, but only returns two versions :facepalm: